### ISSUE TITLE
Review correctness of integer division in floating point context

### ISSUE THREAD

**Issue by dekobon** [joyent/java-manta#479]
See: 

[AesCbcCipherDetails.java:124](https://github.com/joyent/java-manta/blob/master/java-manta-client-unshaded/src/main/java/com/joyent/manta/client/crypto/AesCbcCipherDetails.java#L124)

**Comment by nairashwin952013** [comment]
This was an extremely fun bug to verify since it's nature was mathematical, I also got a chance to use the [Wolfram-Alpha-Calculator](https://www.wolframalpha.com/input/?i=floor(0.0625) which is a fun-tool like ```MATLAB``` to perform complex mathematical operations. To surmise, ```Math.floor()``` works perfectly fine since its returns the *lesser than closest integer towards +infinity* which is essentially required to calculate the ```blocks``` for ```contentLength``` in ```AESCBCCiphers``` while running [AesCbcCipherDetailsTest#calculateContentLengthWorks()](https://github.com/joyent/java-manta/blob/master/java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/AesCbcCipherDetailsTest.java#L98).
I understand why you reported this since the aforementioned function is a bit generic nevertheless we can safely **close** Joyent/java-manta#479.

**Comment by nairashwin952013** [comment]
Although, I suggest I should come up with a pull-request to define the test-group:```unlimited-crypto``` in the ```testing-it``` file.

**Comment by dekobon** [comment]
Ok, I just took a look at this. This is interesting:
```java
    public static void main(String[] argv) {
        int a = -902342342;
        int b = 3233337;
        int intDivision = a / b;
        double floor = Math.floor(a / b);
        double floorDiv = Math.floorDiv(a, b);
        double casted = (double)a / (double)b;
        double floorCasted = Math.floor(casted);

        System.out.printf("a / b as int = %d\n", intDivision);
        System.out.printf("floor(a / b) as double = %f\n", floor);
        System.out.printf("floorDiv(a / b) as double = %f\n", floorDiv);
        System.out.printf("floor((double)a / (double)b) as double = %f\n", floorCasted);
    }
```

Output:
```
a / b as int = -279
floor(a / b) as double = -279.000000
floorDiv(a / b) as double = -280.000000
floor((double)a / (double)b) as double = -280.000000
```
All three values are the same if `a` and `b` are positive. In our case, `Math.floorDiv()` might make the most sense because it converts the ints to a double and does the floor operation all at once.

What are your thoughts?

**Comment by nairashwin952013** [comment]
I agree, Also within this [code](https://github.com/joyent/java-manta/blob/master/java-manta-client-unshaded/src/main/java/com/joyent/manta/client/crypto/AesCbcCipherDetails.java#L124) ```context (AesCbcCipherDetails.java#L124)``` the return type for ```Math.floorDiv()``` would be ```long``` which is what we need here.
The aforementioned approach would also reduce the numbers of ```type-casting``` we perform supplemented by the fact that it's a relatively more accurate method of performing this mathematical operation and checks for ```Overflow-Error```.

**Comment by nairashwin952013** [comment]
Fixed by Pull-Request:#500 .
